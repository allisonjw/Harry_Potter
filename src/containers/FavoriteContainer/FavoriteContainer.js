import React from 'react';
import { connect } from 'react-redux';
import { Favorite } from '../Favorites/Favorite';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import '../FavoriteContainer/FavoriteContainer.scss'; 

export const FavoriteContainer = (props) => {
    const { allCharacters, favoriteCharacters } = props;
    
    const findFavorite = favoriteCharacters.map(id => {
      return allCharacters.find(character => character._id === id);
    });

    const favCharacter = findFavorite.map(character => {
      return <Favorite {...character}/>
    });

    return (
      <section className='section_favorite-background'>
          <section className="section_favorite-heading">
          <Link to='/houses'>
            <button className='back_btn' >Back to Houses</button>
          </Link>
          <h2 className='favorite_heading'>Favorite Characters</h2>
          </section>
          <section className="section_favorite">
            {favCharacter}
          </section>
      </section>
    )
};

export const mapStateToProps = (state) => ({
    allCharacters: state.allCharacters,
    favoriteCharacters: state.favoriteCharacters,
});

export default connect(mapStateToProps)(FavoriteContainer);

FavoriteContainer.propTypes = {
    allCharacters: PropTypes.array,
    favoriteCharacters: PropTypes.array,
}