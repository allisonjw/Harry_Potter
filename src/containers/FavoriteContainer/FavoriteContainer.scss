.section_favorite-background {
    background-image: url(https://www.hp-lexicon.org/wp-content/uploads/2016/05/quidditch_stadium_by_facegrater-d8jpbo1.jpg);
    background-repeat: round;
    height: 100vh;
}

.section_favorite {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    overflow-x: scroll;
    overflow-y: hidden;
    padding: 1%;
}

.section_favorite-heading {
    display: flex;
    flex-direction: row;
    font-size: 2em;
    height: 20%;
    justify-content: space-evenly;
    margin-bottom: 3%;
    width: 100%;
      .favorite_heading {
          margin-right: 15%;
          margin-top: 3%;
      }
}

.back_btn {
    border-radius: 3px;
    font-size: 1.2em;
    height: 60px;
    width: 100%;
    margin: 19% 0 0 15%;
    &:hover {
        cursor: pointer;
        transform: scale(1.1);
        transition: all .4s ease-in-out;
      }
}

