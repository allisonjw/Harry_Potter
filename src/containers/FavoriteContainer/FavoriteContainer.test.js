import React from 'react';
import { FavoriteContainer, mapStateToProps } from './FavoriteContainer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('FavoriteContainer', () => {

    describe('favorites', () => {

       let mockFavoriteCharacter = [
          { 
            _id: "5a0fa54aae5bc100213c232f",
            name: "Bathsheda Babbling",
            role: "Professor, Ancient Runes",
            school: "Hogwarts School of Witchcraft and Wizardry",
            __v: 0,
            ministryOfMagic: false,
            orderOfThePhoenix: false,
            dumbledoresArmy: true,
            deathEater: false,
            bloodStatus: "unknown",
            species: "human"
         }
       ];

       let mockAllCharacters = [
          {
            _id: "5a0fa5deae5bc100213c2330",
            name: "Ludo Bagman",
            role: "Head, Department of Magical Games and Sports",
            __v: 0,
            ministryOfMagic: true,
            orderOfThePhoenix: false,
            dumbledoresArmy: false,
            deathEater: false,
            bloodStatus: "unknown",
            species: "human"
          },
          {   
            _id: "5a0fa54aae5bc100213c232f",
            name: "Bathsheda Babbling",
            role: "Professor, Ancient Runes",
            school: "Hogwarts School of Witchcraft and Wizardry",
            __v: 0,
            ministryOfMagic: false,
            orderOfThePhoenix: false,
            dumbledoresArmy: true,
            deathEater: false,
            bloodStatus: "unknown",
            species: "human"
          } 
       ];

       let wrapper = shallow( <FavoriteContainer 
               favoriteCharacters={mockFavoriteCharacter}
                allCharacters={mockAllCharacters}
       /> )
       
      it('should match snapshot with all information passing in correctly', () => {
         expect(wrapper).toMatchSnapshot();
      })
    })

    describe('mapStateToProps', () => {
        it('should return a allCharacters and favoriteCharacters object', () => {
            const mockState = {
               "favoriteCharacters": [   
                    { 
                        _id: "5a0fa54aae5bc100213c232f",
                        name: "Bathsheda Babbling",
                        role: "Professor, Ancient Runes",
                        school: "Hogwarts School of Witchcraft and Wizardry",
                        __v: 0,
                        ministryOfMagic: false,
                        orderOfThePhoenix: false,
                        dumbledoresArmy: true,
                        deathEater: false,
                        bloodStatus: "unknown",
                        species: "human"
                    }
                ],
               "allCharacters": [
                    { 
                        _id: "5a0fa54aae5bc100213c232f",
                        name: "Bathsheda Babbling",
                        role: "Professor, Ancient Runes",
                        school: "Hogwarts School of Witchcraft and Wizardry",
                        __v: 0,
                        ministryOfMagic: false,
                        orderOfThePhoenix: false,
                        dumbledoresArmy: true,
                        deathEater: false,
                        bloodStatus: "unknown",
                        species: "human"
                    }
               ],
            }
            const expected = {
               "favoriteCharacters": [
                    {
                        _id: "5a0fa54aae5bc100213c232f",
                        name: "Bathsheda Babbling",
                        role: "Professor, Ancient Runes",
                        school: "Hogwarts School of Witchcraft and Wizardry",
                        __v: 0,
                        ministryOfMagic: false,
                        orderOfThePhoenix: false,
                        dumbledoresArmy: true,
                        deathEater: false,
                        bloodStatus: "unknown",
                        species: "human"
                    }
               ],
                "allCharacters": [
                    { 
                        _id: "5a0fa54aae5bc100213c232f",
                        name: "Bathsheda Babbling",
                        role: "Professor, Ancient Runes",
                        school: "Hogwarts School of Witchcraft and Wizardry",
                        __v: 0,
                        ministryOfMagic: false,
                        orderOfThePhoenix: false,
                        dumbledoresArmy: true,
                        deathEater: false,
                        bloodStatus: "unknown",
                        species: "human"
                    }
                ],
            }
            const mappedProps = mapStateToProps(mockState);
            expect(mappedProps).toEqual(expected);
        });
    });
});