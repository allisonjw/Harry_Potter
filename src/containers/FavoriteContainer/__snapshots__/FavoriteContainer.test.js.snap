// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteContainer favorites should match snapshot with all information passing in correctly 1`] = `
<section
  className="section_favorite-background"
>
  <section
    className="section_favorite-heading"
  >
    <Link
      to="/houses"
    >
      <button
        className="back_btn"
      >
        Back to Houses
      </button>
    </Link>
    <h2
      className="favorite_heading"
    >
      Favorite Characters
    </h2>
  </section>
  <section
    className="section_favorite"
  >
    <Favorite />
  </section>
</section>
`;
